File: tournaments/boxBlur/boxBlur.cs
int[][] boxBlur(int[][] image) {
    var rows = image.Length;
    var cols = image[0].Length;
    var result = new int[rows - 2][];

    for (int i = 1; rows - 1 > i; ++i) {
        result[i - 1] = new int[cols - 2];

        for (int j = 1; cols - 1 > j; ++j) {
            result[i - 1][j - 1] = getPixel(image, i, j);
        }
    }

    return result;
}

int getPixel(int[][] image, int row, int col) {
    int result = 0;

    result += image[row - 1][col - 1];
    result += image[row - 1][col];
    result += image[row - 1][col + 1];

    result += image[row][col - 1];
    result += image[row][col];
    result += image[row][col + 1];

    result += image[row + 1][col - 1];
    result += image[row + 1][col];
    result += image[row + 1][col + 1];

    return result/9;
}

File: tournaments/boxBlur/boxBlur.js
function boxBlur(image) {
    const getPixel = function(row, col) {
        let r = 0;

        for (let i = -1; 1 >= i; ++i) {
            for (let j = -1; 1 >= j; ++j) {
                r += image[row + i][col + j];
            }
        }

        return ~~(r / 9);
    };

    const [rows, cols] = [image, image[0]].map(arr => arr.length);
    const result = Array(rows - 2).fill([]).map(_ => Array(cols - 2).fill(0));

    for (let row = 1; rows - 1 > row; ++row) {
        for (let col = 1; cols - 1 > col; ++col) {
            result[row - 1][col - 1] = getPixel(row, col);
        }
    }

    return result;
}
